rancher2/resource_rancher2_cluster.go

```diff
@@ -321,6 +321,18 @@ func resourceRancher2ClusterUpdate(d *schema.ResourceData, meta interface{}) err
 		}
 	}
 
+	// imported_config is special because it applies to *any imported cluster*,
+	// including hosted providers (AKS, EKS, GKE) where `imported = true` is set.
```

Comment:
I am not sure, if we are mixing here settings. For hosted providers `imported` flag is set inside hosted_provider_spec, e.g. https://github.com/rancher/aks-operator/blob/main/pkg/apis/aks.cattle.io/v1/types.go#L43 

Even if hosted provider imported cluster is set, then it should be managed by hoster provider operator

Reply:
sorry, it looks good to me, I had to understand bug description :+1: